File: backends/ruby-hanami/app/domain/invitation/value_objects.rb
# frozen_string_literal: true
#
# 招待の値オブジェクトを定義するドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Invitation
    # 招待情報の値オブジェクトです。
    Vo = Struct.new(:token, :url, :display_url, keyword_init: true)
  end
end

File: backends/ts-hono/src/domain/invitation/valueObjects.ts
/**
 * 招待ドメイン 値オブジェクトモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */

/** 招待トークン取得・発行結果の VO。 */
export interface InvitationResult {
  token: string;
  url: string;
  displayUrl: string;
}
